### PR TITLE
Refactor bluemix to cloud.ibm

### PR DESCRIPTION
**Pull request by mediumTaj**
This pull request refactors bluemix to cloud.ibm. `getCredentialsFromBluemix()` is a private method. Are we safe refactoring this to `getCredentialsFromIbmCloud`?

### CODE REVIEW

**Comment by joyychang** [comment]
You may need to do a search in node-sdk since the base_service is extended, and i'll expect anything that extends it to be able to call `this.functionName` in the parent.

**Comment by mediumTaj** [comment]
OK so the service can still call `getCredentialsFromBluemix()` even though its `private`? No concept of `protected`?

**Comment by mediumTaj** [comment]
It doesn't look like there are any references to `getCredentialsFromBluemix()` in the Watson SDK. Is changing this considered breaking? Can anybody call this method or only classes that extend `base_service`?

**Comment by joyychang** [comment]
I experimented, and the answer is yes.
You would instantiate an instance of the extending class, or base_service itself. then you can do a `instanceName.getCredentialsFromBluemix()`

File: lib/base_service.ts [hunk 1]
```diff
@@ -294,11 +294,11 @@ export class BaseService {
       _options = extend(_options, options);
       return _options;
     }
-    // Get credentials from environment properties or Bluemix,
+    // Get credentials from environment properties or IBM Cloud,
     // but prefer credentials provided programatically
     _options = extend(
       {},
-      this.getCredentialsFromBluemix(this.name),
+      this.getCredentialsFromIbmCloud(this.name),
```
Comment (germanattanasio) [inline lib/base_service.ts]: `IBM` not `Ibm`. I would call it `FromCloud`

**Comment by mediumTaj** [comment]
@germanattanasio I changed the method name to `getCredentialsFromCloud()`.

@joyychang Is it a bad idea to change the method name then? I don't know node - can I keep two methods `getCredentialsFromBluemix()` and `getCredentialsFromCloud()` for compatibility?

File: lib/base_service.ts (hunk 1 above)
Comment (joyychang) [reply lib/base_service.ts]: do we need some sort of approval for this nomenclature?

**Comment by joyychang** [comment]
In this case you may have to, and add a deprecation message for the Bluemix one. @germanattanasio should this be consider breaking then?

**Comment by mediumTaj** [comment]
I can add a deprecation message for the bluemix operation. I'd like @dpopp07 to look at this before merging as well just to make sure everything looks ok. There isn't a lot of urgency in merging this since we are targeting next regular release for these changes.

**Comment by mediumTaj** [comment]
Hmm its not clear to me how to add the deprecation message. In the repo I see deprecation in the inline documentation
```
/**
 * IBM Watson&trade; Language Translator translates text from one language to another. The service offers multiple domain-specific models that you can customize based on your unique terminology and language. Use Language Translator to take news from across the globe and present it in your language, communicate with your customers in their own language, and more.
 * 
 * @deprecated Language Translator v3 is now available. The v2 Language Translator API will no longer be available after July 31, 2018. To take advantage of the latest service enhancements, migrate to the v3 API. View the [Migrating to Language Translator v3](https://console.bluemix.net/docs/services/language-translator/migrating.html) page for more information.
 */
```

and as a console warning

```
if (!options['silent']) {
    // eslint-disable-next-line no-console
    console.warn(
    'WARNING: Conversation V1 is deprecated and will be removed in the next major release of the SDK. Use Assistant V1 or Assistant V2.' +
        ' Set {silent: true} to disable this message.'
    );
}
```

Which is preferred?

**Comment by joyychang** [comment]
You will need to do both. javascript doesn't have a compiler that parse through annotations to make them have meaning. I don't think typescript is any different.

**Pull request merged**